akka-http2-support/src/main/scala/akka/http/impl/engine/http2/Http2ServerDemux.scala

```diff
@@ -106,18 +105,25 @@ class Http2ServerDemux extends GraphStage[BidiShape[Http2SubStream, FrameEvent,
       private var totalOutboundWindowLeft = Http2Protocol.InitialWindowSize
       private var streamLevelWindow = Http2Protocol.InitialWindowSize
 
-      def lastStreamId: Int =
-        incomingStreams.lastOption.map(_._1).getOrElse(0) // FIXME: this is a Map which has no order, so lastStreamId will be an arbitrary value
+      /**
+       * The "last peer-initiated stream that was or might be processed on the sending endpoint in this connection"
+       * @see http://httpwg.org/specs/rfc7540.html#rfc.section.6.8
+       */
+      def lastStreamId: Int = {
+        incomingStreams.keys.toList.sortBy(-_).headOption.getOrElse(0) // FIXME should be optimised
+      }
```

Comment:
well basically I think we end up saving the last streamId as var at some point and even when not the incomingStreams list will probably never hold many keys (depends on stream settings and max stream settings)